### PR TITLE
[UI] Bulk Delete in Filters + Patterns + Applications

### PR DESCRIPTION
**Pull request by Chadha93**
Signed-off-by: Gaurav Chadha <gauravchadha1676@gmail.com>

**Description**

This PR fixes # my delayed issues (apologies) 🙏

**Notes for Reviewers**

https://user-images.githubusercontent.com/65453826/130494357-106cc35f-d776-4b3b-b6b7-54b5ec77a186.mp4

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — CHANGES_REQUESTED** [review]
Removing the confirmation dialog regresses system behavior. We need to ensure that users are prompted on single filter or bulk filter delete.

**Comment by Chadha93** [comment]
@leecalcote Updated, Now the user will be prompted whenever deleting one or more pattern/filter/app(s) 

https://user-images.githubusercontent.com/65453826/130677781-8b8a6715-ce34-4134-93ed-a74915660dd4.mp4

File: ui/components/Filters.js [hunk 1]
```diff
@@ -374,17 +373,18 @@ function MesheryFilters({
...
-      title : "Delete Filters?",
+  async function showmodal() {
+    let response = await modalRef.current.show({ title : "Delete Filter?",
 
       subtitle : "Are you sure you want to delete this filter?",
 
-      options : ["YES", "NO"],
-    });
-    if (response === "NO") return;
+      options : ["YES", "NO"], })
```
Comment (leecalcote) [inline ui/components/Filters.js:381]:
```suggestion
      options : ["yes", "no"], })
```

File: ui/components/Filters.js [hunk 2]
```diff
@@ -429,11 +427,16 @@ function MesheryFilters({
...
-      const fid = Object.keys(row.lookup).map((idx) => filters[idx]?.id);
-      fid.forEach((fid) => deleteFilter(fid));
+    onRowsDelete : async function handleDelete(row) {
+      let response  = await showmodal()
+      console.log(response)
+      if (response === "YES") {
+        const fid = Object.keys(row.lookup).map((idx) => filters[idx]?.id);
+        fid.forEach((fid) => deleteFilter(fid));
+      }
+      if (response === "NO")
```
Comment (leecalcote) [inline ui/components/Filters.js:437]:
```suggestion
      if (response === "no")
```

File: ui/components/Filters.js [hunk 3]
```diff
@@ -429,11 +427,16 @@ function MesheryFilters({
     download : false,
     customToolbar : CustomToolbar(uploadHandler, urlUploadHandler),
 
-    onRowsDelete : function handleDelete(row) {
-      const fid = Object.keys(row.lookup).map((idx) => filters[idx]?.id);
-      fid.forEach((fid) => deleteFilter(fid));
+    onRowsDelete : async function handleDelete(row) {
+      let response  = await showmodal()
+      console.log(response)
+      if (response === "YES") {
```
Comment (leecalcote) [inline ui/components/Filters.js:433]:
```suggestion
      if (response === "yes") {
```

File: ui/components/MesheryApplications.js [hunk 4]
```diff
@@ -373,16 +373,16 @@ function MesheryApplications({
...
-  async function deleteApplication(id) {
-    let response = await modalRef.current.show({
-      title : "Delete Application?",
+  async function showModal() {
+    let response = await modalRef.current.show({ title : "Delete Aplication?",
 
       subtitle : "Are you sure you want to delete this application?",
 
-      options : ["yes", "no"],
+      options : ["YES", "NO"], })
```
Comment (leecalcote) [inline ui/components/MesheryApplications.js:381]:
```suggestion
      options : ["yes", "no"], })
```

File: ui/components/MesheryApplications.js [hunk 5]
```diff
@@ -427,9 +426,15 @@ function MesheryApplications({
     download : false,
     customToolbar : CustomToolbar(uploadHandler),
 
-    onRowsDelete : function handleDelete(row) {
-      const fid = Object.keys(row.lookup).map(idx => applications[idx]?.id)
-      fid.forEach(fid => deleteApplication(fid))
+    onRowsDelete : async function handleDelete(row) {
+      let response = await showModal()
+      console.log(response)
+      if (response === "YES") {
```
Comment (leecalcote) [inline ui/components/MesheryApplications.js:432]: "yes"
```suggestion
      if (response === "yes") {
```

File: ui/components/MesheryApplications.js [hunk 6]
```diff
@@ -427,9 +426,15 @@ function MesheryApplications({
...
+        const fid = Object.keys(row.lookup).map(idx => applications[idx]?.id)
+        fid.forEach(fid => deleteApplication(fid))
+      }
+      if (response === "NO")
```
Comment (leecalcote) [inline ui/components/MesheryApplications.js:436]:
```suggestion
      if (response === "no")
```

File: ui/components/MesheryPatterns.js [hunk 7]
```diff
@@ -443,16 +443,16 @@ function MesheryPatterns({
...
-  async function deletePattern(id) {
-    let response = await modalRef.current.show({
-      title : "Delete Pattern?",
+  async function showModal() {
+    let response = await modalRef.current.show({ title : "Delete Pattern?",
 
       subtitle : "Are you sure you want to delete this pattern?",
 
-      options : ["yes", "no"],
+      options : ["YES", "NO"], })
```
Comment (leecalcote) [inline ui/components/MesheryPatterns.js:451]:
```suggestion
      options : ["yes", "no"], })
```

File: ui/components/MesheryPatterns.js [hunk 8]
```diff
@@ -497,11 +495,18 @@ function MesheryPatterns({
...
-      const fid = Object.keys(row.lookup).map(idx => patterns[idx]?.id)
-      fid.forEach(fid => deletePattern(fid))
+    onRowsDelete : async function handleDelete(row) {
+      let response = await showModal()
+      console.log(response)
+      if (response === "YES") {
+        const fid = Object.keys(row.lookup).map(idx => patterns[idx]?.id)
+        fid.forEach(fid => deletePattern(fid))
+      }
+      if (response === "NO")
```
Comment (leecalcote) [inline ui/components/MesheryPatterns.js:508]:
```suggestion
      if (response === "no")
```

File: ui/components/MesheryPatterns.js [hunk 9]
```diff
@@ -497,11 +495,18 @@ function MesheryPatterns({
...
+
     onCellClick : (_, meta) => meta.colIndex !== 3 && setSelectedRowData(patterns[meta.rowIndex]),
 
-    onRowsDelete : function handleDelete(row) {
-      const fid = Object.keys(row.lookup).map(idx => patterns[idx]?.id)
-      fid.forEach(fid => deletePattern(fid))
+    onRowsDelete : async function handleDelete(row) {
+      let response = await showModal()
+      console.log(response)
+      if (response === "YES") {
```
Comment (leecalcote) [inline ui/components/MesheryPatterns.js:504]:
```suggestion
      if (response === "yes") {
```

**Review by leecalcote — APPROVED** [review]

**Review by navendu-pottekkat — APPROVED** [review]

**Pull request merged**